src/1.generic-constraints/index.ts

```diff
@@ -5,6 +5,26 @@
 
 // ------- start coding here -------
 
+function mapObjectProperties<T, M extends { [key: string]: keyof T }>(
+  source: T,
+  mapping: M
+): {
+  [K in keyof M]: M[K] extends keyof T ? T[M[K]] : never;
```

Comment:
```suggestion
  [K in keyof M]: M[K] extends keyof T ? T[M[K]] : never;
```
Untuk kode ini silahkan dibuat menggunakan sebuah type sendiri yang nantinya akan dipanggil dalam function atau tidak dibuat secara literal.
Tips: pada type yang sudah dibuat, silahkan menggunakan `-readonly` untuk menghilangkan readonly